sky/clouds/azure.py

```diff
@@ -250,13 +250,13 @@ def get_feasible_launchable_resources(self, resources):
 
         def failover_disk_tier(instance_type: str,
                                disk_tier: Optional[str]) -> Optional[str]:
-            if disk_tier is None:
-                ok, _ = Azure.check_disk_tier(instance_type,
-                                              clouds.Cloud._DEFAULT_DISK_TIER)
-                if not ok:
-                    # Auto failover to low disk tier when disk tier
-                    # are not specified
-                    return 'low'
+            if disk_tier:
```

Comment:
style: explicitly check the None value
```suggestion
            if disk_tier is not None:
```